### PR TITLE
Enable (almost) all conformance test on PVC for L0

### PR DESCRIPTION
**Pull request by igchor**
Fix urEnqueueKernelLaunchUSMLinkedList by providing required alignment and calling urKernelSetExecInfo.

Also, skip InvalidKernelArgs for L0 as L0
cannot check kernel arguments.

### CODE REVIEW

File: test/conformance/enqueue/urEnqueueKernelLaunch.cpp [hunk 1]
```diff
@@ -139,8 +139,9 @@ TEST_P(urEnqueueKernelLaunchTest, InvalidKernelArgs) {
                                      nullptr));
 
     if (backend == UR_PLATFORM_BACKEND_CUDA ||
-        backend == UR_PLATFORM_BACKEND_HIP) {
-        GTEST_FAIL() << "AMD and Nvidia can't check kernel arguments.";
+        backend == UR_PLATFORM_BACKEND_HIP ||
+        backend == UR_PLATFORM_BACKEND_LEVEL_ZERO) {
+        GTEST_FAIL() << "AMD, L0 and Nvidia can't check kernel arguments.";
```
Comment (pbalcer) [inline test/conformance/enqueue/urEnqueueKernelLaunch.cpp:144]: sooo... this is a native cpu adapter test then?:D

**Review by pbalcer — APPROVED** [review]

File: test/conformance/enqueue/urEnqueueKernelLaunch.cpp (hunk 1 above)
Comment (igchor) [reply test/conformance/enqueue/urEnqueueKernelLaunch.cpp:144]: Yeah, I guess. Tbh, I don't know if there are any plans to add support for that for other backends.

**Comment by igchor** [comment]
There are a few new test failures (random) for the legacy L0 adapter after recent in cts_exe. I'm wondering if we should add some matches for a specific devices (PVC)?

**Comment by pbalcer** [comment]
> There are a few new test failures (random) for the legacy L0 adapter after recent in cts_exe (most likely a driver issue). I'm wondering if we should add some matches for a specific devices (PVC)?

Yeah, generally if a feature is unsupported the test should skip the test, not fail. This should take care of expected platform differences. However, if there are some bugs that occur only on one platform, then yes, we should add matches with `OPT`.

**Comment by igchor** [comment]
> > There are a few new test failures (random) for the legacy L0 adapter after recent in cts_exe (most likely a driver issue). I'm wondering if we should add some matches for a specific devices (PVC)?
> 
> Yeah, generally if a feature is unsupported the test should skip the test, not fail. This should take care of expected platform differences. However, if there are some bugs that occur only on one platform, then yes, we should add matches with `OPT`.

Done.

File: source/adapters/level_zero/usm_p2p.cpp [hunk 2]
```diff
@@ -70,6 +70,6 @@ ur_result_t urUsmP2PPeerAccessGetInfoExp(ur_device_handle_t commandDevice,
   }
   }
 
-  return ReturnValue(propertyValue);
+  return ReturnValue(uint32_t{propertyValue});
```
[the remark this answers is not in the document]
Comment (pbalcer) [reply source/adapters/level_zero/usm_p2p.cpp:73]: I merged #2246, please update this PR to remove this change

[the remark this answers is not in the document]
Comment (igchor) [reply source/adapters/level_zero/usm_p2p.cpp]: Done. Reverted this change.

**Review by pbalcer — APPROVED** [review]

**Pull request merged**